src/java.base/share/classes/sun/net/www/protocol/file/FileURLConnection.java

```diff
@@ -222,4 +229,17 @@ public Permission getPermission() throws IOException {
         }
         return permission;
     }
+
+    /**
+     * Throw {@link MalformedURLException} if the FTP fallback feature for non-local
+     * file URLs is not explicitly enabled via system property.
+     *
+     * @see #FTP_FALLBACK_ENABLED
+     * @throws MalformedURLException if FTP fallback is not enabled
+     */
+     static void requireFtpFallbackEnabled() throws MalformedURLException {
+        if (!FTP_FALLBACK_ENABLED) {
+            throw new MalformedURLException("Unsupported non-local file URL");
```

Comment:
I aimed to keep this error message terse but informative, without revealing any parts of the URL spec.

I user stumbling on a rejected URL will not have a lot to work from here though..

If anyone have suggestions to improve the message, please chime in!

Reply:
I have some ideas, but you may need to wait until https://git.openjdk.org/jdk/pull/23929 is integrated.